app/jobs/super_good/solidus_taxjar/backfill_transaction_sync_batch_job.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module SuperGood
+  module SolidusTaxjar
+    class BackfillTransactionSyncBatchJob < ApplicationJob
+      queue_as { SuperGood::SolidusTaxjar.job_queue }
+
+      def perform(transaction_sync_batch)
+        ::Spree::Order.complete.where(shipment_state: 'shipped').find_each do |order|
```

Comment:
I am trying to think if there is a more forward compatible way to do the order query 🤔 We know we are going to be adding filtering at the batch level, so what do you think about pre-building the `TransactionSyncLog` records on the batch before queueing the job and fetching the orders through those records 🤔

Reply:
I'm hesitant to do much in the ways of forward compatible! We are adding date filtering at the batch level, but what will that look like?

Maybe it looks like a "start_time" and "end_time" column on the `transaction_sync_batch`, in which case this query could just be modified to be like `::Spree::Order.complete.where(shipment_state: 'shipped', created_at: [transaction_sync_batch.start_time..transaction_sync_batch.end_time])` which seems pretty forward compatible to me! But there could be another implementation, and I'd be hesitant to try and make this work better for an implementation we don't know the shape of yet!

Maybe I'm misunderstanding what you are meaning by filtering

Reply:
It's been so long I already forgot what I was going for here 😂 Let's leave this for now as is and we can refactor when the right time comes 👍🏼